.dockerignore

```diff
@@ -4,6 +4,6 @@
 /aap-azurerm/
 /installerstore
 /tools/
-/server/
```

Comment:
Shouldn't be including source folders in docker context as we build them in separately in make files.
Same for line below.

Reply:
We removed this in order to build a container image from this source. Good discussion point, since without the server and ui being available, no one can build a container image from this source without jumping through hoops

Reply:
I thought we have fixed the make files very early in our collaboration to make sure they run on Macs as well. If there still are issues, we should address them.
With make file, the following commands build the server and the UI and put built binary and folder `public` inside `build/` folder.
```sh
$ make build-server
Building installer server
make build -C server
make[1]: Entering directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/server'
Using go version go1.19.4 linux/amd64
mkdir -p build && go build -o build/server .
make[1]: Leaving directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/server'
cp server/build/server build
```
```sh
make build-web-ui
Building installer web UI
make build -C ui
make[1]: Entering directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/ui'
make nvm CMD="npm run build"
make[2]: Entering directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/ui'
. /home/tznamena/.nvm/nvm.sh --no-use && nvm install && npm i && npm run build
Found '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/ui/.nvmrc' with version <v18.13>
v18.13.0 is already installed.
Now using node v18.13.0 (npm v8.19.3)

up to date, audited 1306 packages in 3s

232 packages are looking for funding
  run `npm fund` for details

7 high severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
npm notice 
npm notice New major version of npm available! 8.19.3 -> 9.6.6
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.6.6
npm notice Run npm install -g npm@9.6.6 to update!
npm notice 

> aap-installer@0.0.1 build
> react-scripts build

Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  136.94 kB  build/static/js/main.77e75936.js
  59.13 kB   build/static/css/main.9a3d4d12.css

The project was built assuming it is hosted at /.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.
You may serve it with a static server:

  npm install -g serve
  serve -s build

Find out more about deployment here:

  https://cra.link/deployment

make[2]: Leaving directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/ui'
make[1]: Leaving directory '/home/tznamena/Projects/Ansible/AoC/Code/azure-aap-deployment-driver/ui'
cp -ap ui/build/. build/public
```

Reply:
Agreed, we were just relying 

> Did a quick pass through this and I noticed some of the existing stuff being duplicated (builds, caddy, docker compose,...). If this is just to get us started that's ok, I can contribute changes to use existing components.
> 
> Please let me know if you found anything that existing components could not handle.

This is just to get started...

Reply:
We'll remove the `build.sh` as part of this PR (makes sense), and move the `build-modm.sh` into ./package (just temporarily) so we can build modm binaries into the container image with all three binaries.